### DISCUSSION TITLE
Restrict +- (additions, subpractions) for multiple object field in Api Rules

### DISCUSSION THREAD

**Discussion by pranjaldugar** [pocketbase/pocketbase#6195] [Q&A]
Hi there,
In API Rules,
When a field is of type multiple (relations, selections, files etc)
how can I restrict to only allow additions to a field not subpractions?
Something like this is not working
```
@request.data.books-:isset = false
```

**Comment by ganigeorgiev** [comment]
It is not possible because the result of `@request.data.*` (`@request.body.*` in v0.23+) is replaced with the resolved value. It is not clear what exactly is the use case or the desired behavior but a workaround could be to use the [`:length` modifier](https://pocketbase.io/docs/api-rules-and-filters/#length-modifier) and compare against the old/existing value, aka. `@request.data.books:isset = false || books:length <= @request.data.books:length`.

**Reply by ganigeorgiev** [reply]
Also note that if `books` is a multi-valued field, the above doesn't prevent the removal of existing values or replacing them with different ones (e.g. `[a,b,c]` could be changed to `[d,e,f,g]` which will match the constraint).

If you want to prevent removal, then you'll need to use the related hooks OR change your collections to a more traditional SQL structure (e.g. using a junction collection with 2 single `relation` field pointing back to the related collections).

**Reply by pranjaldugar** [reply]
The workaround would not work in this case,
Its a rental library,
In transactions table, I want to restrict users to not change a book once its reserved,
they can reserve upto 3 books, so they can reserve it any time before the delivery date,
but not change a book once reserved.

Alternative would be create another relation of sub transactions or write custom script which i can do,
but I think, this same pattern could be useful for others also, so it might be a feature request.
Thx

**Reply by ganigeorgiev** [reply]
I have no plans to change the current behavior of the rules. My recommendation if you want to prevent removal is to use a junction collection as suggested above.
